pkg/cache/clusterqueue.go

```diff
@@ -316,6 +332,7 @@ func (c *ClusterQueue) deleteWorkload(w *kueue.Workload) {
 	if c.podsReadyTracking && !apimeta.IsStatusConditionTrue(w.Status.Conditions, kueue.WorkloadPodsReady) {
 		c.WorkloadsNotReady.Delete(k)
 	}
+	c.AllocatableResourceIncreasedGen++
```

Comment:
Are you saying that if the only change that happen in the snapshot was a workload add, then there is no reason to try the previous flavors again, because we know the workload won't fit. Correct?

Regardless of whether my assumption is correct, please add a comment in the code to explain the rationale.

Reply:
done